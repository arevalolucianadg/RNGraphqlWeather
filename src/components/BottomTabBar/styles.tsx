import styled from 'styled-components/native';

export const BottomTabBarWrapper = styled.View`
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  padding: 15px 0;
`;
