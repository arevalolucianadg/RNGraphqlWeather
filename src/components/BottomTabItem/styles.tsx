import styled from 'styled-components/native';

export const TabItem = styled.View`
  display: flex;
  align-items: center;
  justify-content: center;
`;
export const TabIcon = styled.View`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 14px;
  margin-bottom: 2px;
  width: 14px;
`;

interface TabLabelProps {
  active?: boolean;
}
export const TabLabel = styled.Text<TabLabelProps>`
  color: ${props => (props.active ? '#504ED3' : '#828282')};
  font-weight: ${props => (props.active ? 600 : 500)};
`;
