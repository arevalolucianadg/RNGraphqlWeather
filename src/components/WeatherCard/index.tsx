import React, {useState} from 'react';
import {ListRenderItemInfo, View} from 'react-native';
import {useNavigation} from '@react-navigation/native';

import {
  CityName,
  Info,
  WeatherCardWrapper,
  CityTemp,
  CitySummaryTitle,
  WeatherIcon,
} from './WeatherCard.styles';

import ICON from '../../assets/png/weatherIcons';
import { WeatherInfo } from '../../graphql/interfaces';

interface WeatherCardProps {
  city: ListRenderItemInfo<WeatherInfo>;
}

const WeatherCard = ({city: {item}}: WeatherCardProps) => {
  
  const [isFavorite, setIsFavorite] = useState(false);
  
  const {id, name, weather: {summary, temperature}} = item;

  const navigation = useNavigation();

  return (
    <WeatherCardWrapper
      isFavorite={isFavorite}
      onPress={() => navigation.navigate('WeatherDetail', {cityId: id})}>
      <View>
        <CityName isFavorite={isFavorite}>{name}</CityName>
        <Info>
          <CityTemp isFavorite={isFavorite}>{temperature.actual.toFixed(1)}Â°</CityTemp>
          <CitySummaryTitle isFavorite={isFavorite}>
            {summary.title}
          </CitySummaryTitle>
        </Info>
      </View>
      <WeatherIcon source={ICON[`icon${summary.icon}`]} />
    </WeatherCardWrapper>
  );
};

export default WeatherCard;
