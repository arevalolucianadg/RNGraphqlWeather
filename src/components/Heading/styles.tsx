import styled from 'styled-components/native';
import {global} from '../../styles/global-styles';

interface TitleTextProps {
  primary?: boolean;
}
export const TitleText = styled.Text<TitleTextProps>`
  color: ${props =>
    props.primary ? props.theme.colors.primary : props.theme.colors.text};
  font-family: ${global.font.weightBold};
  font-size: 32px;
  line-height: 38px;
`;
