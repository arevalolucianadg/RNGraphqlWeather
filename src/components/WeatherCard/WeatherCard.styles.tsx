import styled, {css} from 'styled-components/native';

import {global} from '../../styles/global-styles';

interface IsFavoriteProps {
  isFavorite: boolean;
}
export const WeatherCardWrapper = styled.TouchableOpacity<IsFavoriteProps>`
  background-color: ${props =>
    props.isFavorite ? props.theme.colors.primary : props.theme.colors.paper};
  border-radius: ${global.border.RadiusMd};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  height: 100px;
  margin-bottom: 16px;
  padding: 16px;
`;

const textColor = css<IsFavoriteProps>`
  color: ${props =>
    props.isFavorite ? props.theme.colors.white : props.theme.colors.text};
  font-family: ${global.font.weightMedium};
`;

export const CityName = styled.Text<IsFavoriteProps>`
  ${textColor}
  font-family: ${global.font.weightMedium};
  font-size: 22px;
  margin-bottom: 8px;
`;
export const Info = styled.View`
  display: flex;
  flex-direction: row;
  align-items: flex-end;
`;
export const CityTemp = styled.Text<IsFavoriteProps>`
  ${textColor}
  font-family: ${global.font.weightBold};
  font-size: 18px;
  letter-spacing: 0.4px;
  margin-right: 8px;
`;
export const CitySummaryTitle = styled.Text<IsFavoriteProps>`
  ${textColor}
  font-family: ${global.font.weightLight};
  font-size: 14px;
`;

export const WeatherIcon = styled.Image`
  height: 70px;
  width: 70px;
`;
