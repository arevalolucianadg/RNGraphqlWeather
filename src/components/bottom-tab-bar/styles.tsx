import { EdgeInsets } from 'react-native-safe-area-context';
import { styled } from '@core/styles';

/**
 * Types
 */

interface BottomTabBarWrapperProps {
  insets: EdgeInsets;
}

/**
 * Components
 */

export const BottomTabBarWrapper = styled.View<BottomTabBarWrapperProps>`
  background-color: ${props => props.theme.colors.paper};
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;

  ${({ insets: { bottom } }) => `padding: 10px 0 ${bottom}px`};
`;
