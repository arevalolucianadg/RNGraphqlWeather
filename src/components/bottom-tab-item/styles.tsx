import { global, styled } from '@core/styles';

export const TabItem = styled.View`
  display: flex;
  align-items: center;
  justify-content: center;
`;
export const TabIcon = styled.View`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 16px;
  margin-bottom: 2px;
  width: 16px;
`;

interface TabLabelProps {
  active?: boolean;
}
export const TabLabel = styled.Text<TabLabelProps>`
  color: ${props =>
    props.active
      ? props.theme.colors.primary
      : props.theme.colors.inactiveIcon};
  font-family: ${props =>
    props.active ? global.font.weightSemiBold : global.font.weightMedium};
`;
