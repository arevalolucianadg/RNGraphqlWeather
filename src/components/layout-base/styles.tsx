import { styled } from '@core/styles';

interface LayoutProps {
  spacing: boolean;
}

export const Layout = styled.View<LayoutProps>`
  ${({ spacing }) => spacing && 'padding: 32px 16px 0'};

  background-color: ${({ theme }) => theme.colors.background};
  flex: 1;
`;
