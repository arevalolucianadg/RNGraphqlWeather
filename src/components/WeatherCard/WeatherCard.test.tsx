import React from 'react';
import {render} from '@testing-library/react-native';
import {ThemeProvider} from 'styled-components/native';

import WeatherCard from './';
import {light} from '../../styles/theme';

const mockedDispatch = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
  };
});

describe('Weather Card', () => {
  const city = {
    id: '2267057',
    name: 'Lisbon',
    weather: {
      summary: {
        icon: '03n',
        title: 'Clouds',
      },
      temperature: {
        actual: 15.81,
      },
    },
  };

  it('Renderiza correctamente', () => {
    const {toJSON} = render(
      <ThemeProvider theme={light}>
        <WeatherCard isFavorite={false} city={city} />
      </ThemeProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('Muestra el nombre de la ciudad', () => {
    const WeatherName = render(
      <ThemeProvider theme={light}>
        <WeatherCard isFavorite={false} city={city} />
      </ThemeProvider>,
    );
    expect(WeatherName.getByText(city.name)).toBeTruthy;
  });
});
