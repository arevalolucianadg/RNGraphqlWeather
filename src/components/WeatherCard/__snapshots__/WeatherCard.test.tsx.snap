// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Card Renderiza correctamente 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F7F7F7",
      "borderRadius": 8,
      "display": "flex",
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "space-between",
      "marginBottom": 16,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <View>
    <Text
      isFavorite={false}
      style={
        Array [
          Object {
            "color": "#333333",
            "fontFamily": "WorkSans-Medium",
            "fontSize": 22,
            "marginBottom": 8,
          },
        ]
      }
    >
      Lisbon
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        isFavorite={false}
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "WorkSans-Bold",
              "fontSize": 18,
              "letterSpacing": 0.4,
              "marginRight": 8,
            },
          ]
        }
      >
        15.8
        Â°
      </Text>
      <Text
        isFavorite={false}
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "WorkSans-Light",
              "fontSize": 14,
            },
          ]
        }
      >
        Clouds
      </Text>
    </View>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/png/weatherIcons/night-cloudy.png",
      }
    }
    style={
      Array [
        Object {
          "height": 70,
          "width": 70,
        },
      ]
    }
  />
</View>
`;
